pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/DeclarativeLinterCommand.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.pipeline.modeldefinition;
+
+import hudson.Extension;
+import hudson.cli.CLICommand;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+import org.apache.commons.io.IOUtils;
+import org.codehaus.groovy.control.Janitor;
+import org.codehaus.groovy.control.MultipleCompilationErrorsException;
+import org.codehaus.groovy.control.messages.SyntaxErrorMessage;
+import org.jenkinsci.plugins.pipeline.modeldefinition.endpoints.ModelConverterAction;
+import org.jenkinsci.plugins.pipeline.modeldefinition.parser.Converter;
+import org.kohsuke.args4j.Argument;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.List;
+
+@Extension
+public class DeclarativeLinterCommand extends CLICommand {
+    @Argument(metaVar="JENKINSFILE", usage="Path to the Jenkinsfile to validate", required=true)
+    String jenkinsfile;
+
+    @Override
+    public String getShortDescription() {
+        return Messages.DeclarativeLinterCommand_ShortDescription();
+    }
+
+    protected int run() throws Exception {
+        File jf = new File(jenkinsfile);
+
+        int retVal = 0;
+        List<String> output = new ArrayList<>();
+
+        String script = null;
+
+        try {
+            FileInputStream fif = new FileInputStream(jf);
```

Comment:
Otherwise you would need to send a FileCallable back to the client, and then it won't work with ssh.